membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -123,6 +125,9 @@ case class RatePlanCharge(
     endDateCondition: EndDateCondition,
     upToPeriods: Option[Int],
     upToPeriodsType: Option[UpToPeriodsType],
+    chargedThroughDate: Option[LocalDate],
+    effectiveStartDate: LocalDate,
+    effectiveEndDate: LocalDate,
```

Comment:
theses dates were on the RatePlan (see old lines 171-174) but in zuora they are actually on the charges.  We were just taking the max chargedThroughDate and the min of the other two across all charges